### PR TITLE
[#9382] Add support for Mcq type questions

### PR DESCRIPTION
**Pull request by RonakLakhotia**
Part of #9382

### CODE REVIEW

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java [hunk 1]
```diff
@@ -37,6 +37,7 @@
     private int numOfMcqChoices;
     private List<String> mcqChoices;
     private boolean otherEnabled;
+    private boolean generatedOptionsEnabled;
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: I don't think you should change this. This would have impact to the DB.

File: src/web/types/default-question-structs.ts [hunk 2]
```diff
@@ -133,8 +133,9 @@ export const DEFAULT_MCQ_QUESTION_DETAILS: () => FeedbackMcqQuestionDetails =
         mcqWeights: [],
         mcqOtherWeight: 0,
         numOfMcqChoices: 0,
-        mcqChoices: [],
+        mcqChoices: [' ', ' '],
```
Comment (wkurniawan07) [inline src/web/types/default-question-structs.ts]: Why is this changed?

File: src/main/java/teammates/common/util/Const.java [hunk 3]
```diff
@@ -445,6 +447,7 @@ private Const() {
 
         // Msq
         public static final int MSQ_MIN_NUM_OF_CHOICES = 2;
+        public static final String EMPTY_MCQ_OPTION = "";
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/util/Const.java]: You shouldn't need this? An empty string is well-understood.

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java [hunk 4]
```diff
@@ -565,6 +567,20 @@ public String getQuestionTypeChoiceOption() {
                         + Const.FeedbackQuestion.MCQ_MIN_NUM_OF_CHOICES + ".");
             }
 
+            // If there are Empty Mcq options entered, at least two of the Mcq options must be filled, given that the
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: Is empty MCQ option allowed in the first place?

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java [hunk 5]
```diff
@@ -598,6 +614,14 @@ public String getQuestionTypeChoiceOption() {
                 if (otherEnabled && mcqOtherWeight < 0) {
                     errors.add(Const.FeedbackQuestion.MCQ_ERROR_INVALID_WEIGHT);
                 }
+
+                // If one of the mcq has a weight assigned but the mcq text is empty trigger this error
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: What's the reason for this? If an empty MCQ string is allowed, there shouldn't be any reason to prevent assigning weight to it?

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    this.model.mcqChoices.push('');
+    if (this.model.hasAssignedWeights) {
+      this.model.mcqWeights.push(0);
+    }
+  }
+
+  /**
+   * Deletes a Mcq.
+   */
+  onMcqDeleted(event: number): void {
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: Applies to other appropriate parts: `onMcq*` or `onMcqOption*`?

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+      questionType: FeedbackQuestionType.MCQ,
+    });
+  }
+
+  /**
+   * Reorders the list on dragging the Mcqs.
+   */
+  onMcqDropped(event: CdkDragDrop<string[]>): void {
+    moveItemInArray(this.model.mcqChoices, event.previousIndex, event.currentIndex);
+    moveItemInArray(this.model.mcqWeights, event.previousIndex, event.currentIndex);
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: What if there is no weight?

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+@Component({
+  selector: 'tm-mcq-question-edit-details-form',
+  templateUrl: './mcq-question-edit-details-form.component.html',
+  styleUrls: ['./mcq-question-edit-details-form.component.scss'],
+})
+export class McqQuestionEditDetailsFormComponent
+    extends QuestionEditDetailsFormComponent<FeedbackMcqQuestionDetails> {
+
+  CURRENT_NUMBER_OF_MCQ_OPTIONS: number = 2;
+  IS_GENERATE_OPTIONS_FROM_LIST_CHECKED: boolean = false;
```
[the remark this answers is not in the document]
Comment (RonakLakhotia) [reply src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: resolved in the latest commit, dont need these variables.

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+ */
+@Component({
+  selector: 'tm-mcq-question-edit-details-form',
+  templateUrl: './mcq-question-edit-details-form.component.html',
+  styleUrls: ['./mcq-question-edit-details-form.component.scss'],
+})
+export class McqQuestionEditDetailsFormComponent
+    extends QuestionEditDetailsFormComponent<FeedbackMcqQuestionDetails> {
+
+  CURRENT_NUMBER_OF_MCQ_OPTIONS: number = 2;
```
[the remark this answers is not in the document]
Comment (RonakLakhotia) [reply src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: don't need this, resolved

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    super({
+      hasAssignedWeights: false,
+      mcqWeights: [],
+      mcqOtherWeight: 1,
+      numOfMcqChoices: 5,
+      mcqChoices: [''],
+      otherEnabled: false,
+      generateOptionsFor: FeedbackParticipantType.NONE,
+      questionText: '',
+      questionType: FeedbackQuestionType.NUMSCALE,
```
[the remark this answers is not in the document]
Comment (RonakLakhotia) [reply src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: resolved in latest commit

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java (hunk 1 above)
Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: I am not sure how to use the checkbox in the frontend otherwise. 
Workflow:-
1) If the user clicks the checkbox, then discards the change, the checkbox is still ticked. Which is why I included this

File: src/main/java/teammates/common/util/Const.java (hunk 3 above)
Comment (RonakLakhotia) [reply src/main/java/teammates/common/util/Const.java]: Will revert it

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts (hunk 6 above)
Comment (RonakLakhotia) [reply src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: Yup `onMcqOption` seems more appropriate

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java (hunk 1 above)
Comment (wkurniawan07) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: > If the user clicks the checkbox

Which checkbox?

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts (hunk 7 above)
Comment (RonakLakhotia) [reply src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: Seems to work fine even when hasAssignedWeights is false. The moveItemInArray implementation seems to take care of empty array cases

**Comment by RonakLakhotia** [comment]
While working on the submission page, I am not sure how to retrieve the Mcq options when generatedOptionsFor is enabled. 
Screenshot 2019-03-12 at 7 20 07 PM
For instance, in the above case I am not sure how to retrieve Team1 and Team3

File: src/web/types/default-question-structs.ts (hunk 2 above)
Comment (RonakLakhotia) [reply src/web/types/default-question-structs.ts]: My bad, will revert this

**Comment by xpdavid** [comment]
You need use API to get a list of students/team from the backend.

File: src/web/services/feedback-questions.service.ts [hunk 11]
```diff
@@ -284,14 +292,40 @@ export class FeedbackQuestionsService {
...
+            questionType: FeedbackQuestionType.MCQ,
+          } as FeedbackMcqQuestionDetails,
           giverType: FeedbackParticipantType.STUDENTS,
           recipientType: FeedbackParticipantType.OWN_TEAM_MEMBERS,
 
           numberOfEntitiesToGiveFeedbackToSetting: NumberOfEntitiesToGiveFeedbackToSetting.UNLIMITED,
 
-          showResponsesTo: [FeedbackVisibilityType.INSTRUCTORS, FeedbackVisibilityType.RECIPIENT,
-            FeedbackVisibilityType.GIVER_TEAM_MEMBERS],
+          showResponsesTo: [FeedbackVisibilityType.INSTRUCTORS, FeedbackVisibilityType.RECIPIENT],
```
[the remark this answers is not in the document]
Comment (RonakLakhotia) [reply src/web/services/feedback-questions.service.ts]: Mistake while deconflicting, reverted

**Comment by RonakLakhotia** [comment]
`GetFeedbackQuestionRecipientsAction` similar to this endpoint?

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java (hunk 1 above)
Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: Screenshot 2019-03-12 at 8 05 39 PM

**Comment by xpdavid** [comment]
I think you can first ignore this case (i.e. when `generatedOptionsFor` is set) and wait for API consolidation to complete. @wkurniawan07 Is that OK with you?

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java (hunk 1 above)
Comment (wkurniawan07) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: Why not make the condition the same as whatever condition makes it unticked in the beginning?

Comment (xpdavid) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: > If the user clicks the checkbox, then discards the change, the checkbox is still ticked.

I would consider it as a bug in V6.

File: src/web/app/components/question-types/question-edit-details-form/weight-field/weight-field.component.html [hunk 12]
```diff
@@ -0,0 +1,5 @@
+<br>
```
[the remark this answers is not in the document]
Comment (RonakLakhotia) [reply src/web/app/components/question-types/question-edit-details-form/weight-field/weight-field.component.html]: Will change this

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java (hunk 1 above)
Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: >Why not make the condition the same as whatever condition makes it unticked in the beginning? 
I am not sure how to listen to event when discard is clicked, and update the checkbox in the Mcq component

Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: @xpdavid the workflow in v6 is fine, I was just using the workflow as an example here

Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: >Why not make the condition the same as whatever condition makes it unticked in the beginning

Fixed this in latest commit

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java (hunk 4 above)
Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: The user can leave the mcq option blank or the case when the user just enters blank spaces in the box

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java (hunk 5 above)
Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: True. Will revert this

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqResponseDetails.java [hunk 13]
```diff
@@ -77,7 +77,21 @@ public String getAnswerCsv(FeedbackQuestionDetails questionDetails) {
...
+        List<String> mcqChoices = ((FeedbackMcqQuestionDetails) correspondingQuestion
+                .getQuestionDetails()).getMcqChoices();
+
+        // if other option is not selected and selected answer is not part of Mcq option list trigger this error.
+        if (!isOther && !mcqChoices.contains(getAnswerString())) {
+            errors.add(getAnswerString() + " " + Const.FeedbackQuestion.MCQ_ERROR_INVALID_OPTION);
+        }
+
+        // if other option is selected but not text is provided trigger this error
+        if (isOther && getAnswerString().trim().equals("")) {
```
Comment (RonakLakhotia) [inline src/main/java/teammates/common/datatransfer/questions/FeedbackMcqResponseDetails.java]: Should the user be allowed to submit a feedback if `isOther` is selected and not text is provided for the other option?

File: src/web/app/components/question-types/question-edit-answer-form/mcq-scale-question-edit-answer-form.component.html [hunk 14]
```diff
@@ -0,0 +1,21 @@
+<tr>
+  <td>
+    <div class="radio" *ngFor="let num of questionDetails.mcqChoices; let i = index;">
+      <label class="margin-right-15px">
+        <input type="radio" class="margin-right-15px" name="check_{{ id }}" [checked]="isMcqOptionSelected[i]"
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-answer-form/mcq-scale-question-edit-answer-form.component.html]: Use `kebab-case` instead of `snake_case`

File: src/web/app/components/question-types/question-edit-answer-form/mcq-scale-question-edit-answer-form.component.html [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+        (click)="updateSelectedMcqOption(i)">
+         <strong>{{ questionDetails.mcqChoices[i] }}</strong>
+      </label>
+    </div>
+    <div class="radio" *ngIf="questionDetails.otherEnabled">
+      <label class="margin-right-15px">
+        <input type="radio" class="margin-right-15px" name="check_{{ id }}" [checked]="responseDetails.isOther"
+        (click)="updateIsOtherOption()">
+        <strong>Random Text Other</strong>
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-answer-form/mcq-scale-question-edit-answer-form.component.html]: This should not be what is shown to the user...?

File: src/web/app/components/question-types/question-edit-details-form/mcq-field/mcq-field.component.html [hunk 16]
```diff
@@ -0,0 +1,14 @@
+<div class="input-group width-100-pc">
+            <span class="input-group-addon">
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-details-form/mcq-field/mcq-field.component.html]: Why this suddenly weird left-indentation?

File: src/web/app/components/question-types/question-edit-details-form/mcq-field/mcq-field.component.html [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+                <span>&#8597; </span>
+                <input type="radio" class="disabled_radio" disabled>
+            </span>
+  <input type="text" class="form-control" [disabled]="!isEditable" [(ngModel)]="text" #mcq
+         (blur)="onMcqOptionEntered(mcq.value)">
+  <span class="input-group-btn">
+              <button type="button" class="btn btn-default input-group-addon" style="cursor: initial" tabindex="-1"
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-details-form/mcq-field/mcq-field.component.html]: Similarly here?

File: src/web/app/components/question-types/question-edit-details-form/mcq-field/mcq-field.component.ts [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+  @Output()
+  elementDeleted: EventEmitter<any> = new EventEmitter();
+
+  @Output()
+  mcqText: EventEmitter<any> = new EventEmitter();
+
+  constructor(private statusMessageService: StatusMessageService) {}
+
+  /**
+   * Deletes a Mcq.
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-details-form/mcq-field/mcq-field.component.ts]: Delete the MCQ or the option?

File: src/web/app/components/question-types/question-edit-details-form/mcq-field/mcq-field.component.ts [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+   */
+  deleteMcqOption(): void {
+    if (this.numberOfMcqChoices > 2) {
+      this.elementDeleted.emit(this.index);
+    } else {
+      this.statusMessageService.showErrorMessage('There must be at least two Mcq options.');
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-details-form/mcq-field/mcq-field.component.ts]: `MCQ` or `Mcq`?
Also, you should standardize the casing. You should not mix up `mcq`, `MCQ`, `Mcq` unless the situation calls, e.g. for variable name `mcq*` or `*Mcq*` is appropriate, but everything else should be standardized.

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+  increaseNumberOfOptions(): void {
+    this.model.numOfMcqChoices += 1;
+    this.model.mcqChoices.push('');
+    if (this.model.hasAssignedWeights) {
+      this.model.mcqWeights.push(0);
+    }
+  }
+
+  /**
+   * Deletes a Mcq.
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: Again, delete the MCQ or the option?

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+   * Displays new mcq weight at specified index.
+   */
+  onMcqWeightEntered(event: number, index: number): void {
+    this.model.mcqWeights[index] = event;
+  }
+
+  /**
+   * Tracks the Mcq option by index.
+   */
+  trackMcqOption(_index: number, item: string[]): string {
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: The variable `index` is used, so there is no reason to prefix it with `_`.

File: src/web/app/components/question-types/question-edit-details-form/weight-field/weight-field.component.ts [hunk 22]
```diff
@@ -0,0 +1,30 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+
+/**
+ * The input field to specify weights for options.
+ */
+@Component({
+  selector: '@tm-weight-field',
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-details-form/weight-field/weight-field.component.ts]: Why the `@`?

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java (hunk 4 above)
Comment (wkurniawan07) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: Then safe to say you want to just prohibit empty options altogether? Otherwise this can be easily gamed with e.g. 2 valid options and 100 empty strings.

Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: > 2 valid options and 100 empty strings.
Exactly. I'll update the validation to prohibit all empty options

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqResponseDetails.java (hunk 13 above)
Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqResponseDetails.java]: What should be the expected behavior for this?

File: src/web/app/components/question-types/question-edit-answer-form/mcq-scale-question-edit-answer-form.component.html (hunk 15 above)
Comment (RonakLakhotia) [reply src/web/app/components/question-types/question-edit-answer-form/mcq-scale-question-edit-answer-form.component.html]: Only Other is meant to be shown.

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqResponseDetails.java (hunk 13 above)
Comment (wkurniawan07) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqResponseDetails.java]: If you want to prohibit empty string as answer, this should be no different.

Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqResponseDetails.java]: Got it. 👍

**Comment by RonakLakhotia** [comment]
Ready for review

File: src/web/app/components/question-types/question-edit-answer-form/mcq-scale-question-edit-answer-form.component.ts [hunk 23]
```diff
@@ -38,14 +38,15 @@ export class McqScaleQuestionEditAnswerFormComponent
...
+   * Updates the other option radio box when clicked.
    */
   updateIsOtherOption(): void {
     this.responseDetails.isOther = !this.responseDetails.isOther;
+    this.isMcqOptionSelected[this.indexOfPreviousOptionSelected] = false;
     if (!this.responseDetails.isOther) {
       this.responseDetails.otherFieldContent = '';
     } else {
-      this.responseDetails.answer = '';
+      this.responseDetails.answer = ' ';
```
Comment (RonakLakhotia) [inline src/web/app/components/question-types/question-edit-answer-form/mcq-scale-question-edit-answer-form.component.ts]: Reverted in the latest commit

File: src/web/app/components/question-types/question-edit-details-form/mcq-field/mcq-field.component.scss [hunk 24]
```diff
@@ -0,0 +1,14 @@
+
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-details-form/mcq-field/mcq-field.component.scss]: Remove this empty line?

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts [hunk 25]
```diff
@@ -0,0 +1,132 @@
...
+  readonly PARTICIPANT_TYPES: string[] = [FeedbackParticipantType.STUDENTS,
+    FeedbackParticipantType.STUDENTS_EXCLUDING_SELF, FeedbackParticipantType.TEAMS,
+    FeedbackParticipantType.TEAMS_EXCLUDING_SELF, FeedbackParticipantType.INSTRUCTORS];
+
+  constructor() {
+    super(DEFAULT_MCQ_QUESTION_DETAILS());
+  }
+
+  ngOnInit(): void {
+    if (this.model.numOfMcqChoices === 0) {
```
Comment (wkurniawan07) [inline src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: Not sure if you should add this here... this is adding an unexpected logic for the component.

Comment (RonakLakhotia) [reply src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: Can I change the Default structure for Mcq question in that case? The idea is to have at least 2 Mcqs displayed to the user when the question is first selected

Comment (wkurniawan07) [reply src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts]: Why do you need to change the default structure? If you need at least 2 MCQs displayed, then modify the model when you're passing it to this component.

File: src/web/app/components/question-edit-form/question-edit-form.component.ts [hunk 26]
```diff
@@ -168,6 +169,18 @@ export class QuestionEditFormComponent implements OnInit {
 
   ngOnInit(): void {}
 
+  /**
+   * Modifies the model for Mcq question.
```
Comment (wkurniawan07) [inline src/web/app/components/question-edit-form/question-edit-form.component.ts]: This is still the same unexpected logic as before, you're only moving it from the other component to here. Why don't just change `model.questionDetails` directly?

File: src/web/services/feedback-questions.service.ts [hunk 27]
```diff
@@ -271,20 +270,33 @@ export class FeedbackQuestionsService {
...
+        };
+
+      case FeedbackQuestionType.MCQ:
+
+        return {
+          questionBrief: '',
+          questionDescription: '',
 
+          questionType: FeedbackQuestionType.MCQ,
+          questionDetails: DEFAULT_MCQ_QUESTION_DETAILS(),
```
Comment (RonakLakhotia) [inline src/web/services/feedback-questions.service.ts]: You mean to change it here?

Comment (wkurniawan07) [reply src/web/services/feedback-questions.service.ts]: Exactly, that way you (and anyone reading the code) know exactly the value being set.

Comment (RonakLakhotia) [reply src/web/services/feedback-questions.service.ts]: Right, I was just skeptical because I thought Default structure had to be used to initialize question details and maintain consistency across the questions

File: src/web/services/feedback-questions.service.ts [hunk 28]
```diff
@@ -296,7 +297,17 @@ export class FeedbackQuestionsService {
           questionDescription: '',
 
           questionType: FeedbackQuestionType.MCQ,
-          questionDetails: DEFAULT_MCQ_QUESTION_DETAILS(),
+          questionDetails: {
```
Comment (RonakLakhotia) [inline src/web/services/feedback-questions.service.ts]: @wkurniawan07 is this fine?

**Comment by RonakLakhotia** [comment]
@wkurniawan07 updated changes
